esphome/components/ethernet_info/ethernet_info_text_sensor.h

```diff
@@ -12,19 +12,31 @@ namespace ethernet_info {
 class IPAddressEthernetInfo : public PollingComponent, public text_sensor::TextSensor {
  public:
   void update() override {
-    auto ip = ethernet::global_eth_component->get_ip_address();
-    if (ip != this->last_ip_) {
-      this->last_ip_ = ip;
-      this->publish_state(network::IPAddress(ip).str());
+    std::string address_results;
+    auto ips = ethernet::global_eth_component->get_ip_addresses();
+    if (ips != this->last_ips_) {
+      this->last_ips_ = ips;
+      this->publish_state(ips[0].str());
+      uint8_t sensor = 0;
+      for (auto &ip : ips) {
+        if (ip.is_set()) {
+          if (ip_sensors_[sensor]) {
```

Comment:
Please alter to explicitly check for `nullptr`
```suggestion
          if (ip_sensors_[sensor] != nullptr) {
```